lib/domain/usecase/room/chat_get_pinned_events_interactor.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:async';
+
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/room/chat_get_pinned_events_state.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatGetPinnedEventsInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Room room,
+  }) async* {
+    Logs().d(
+      "ChatGetPinnedEventsInteractor()::execute()::roomId: ${room.id}",
+    );
+    yield Right(ChatGetPinnedEventsLoading());
+    try {
+      final pinnedEvents = room.pinnedEventIds;
+      final result = await Future.wait(
+        pinnedEvents.map(room.getEventById),
+      );
+
+      if (result.isEmpty) {
+        yield Left(ChatGetPinnedEventsNoResult());
+        return;
+      } else {
+        Logs().d(
+          "ChatGetPinnedEventsInteractor()::execute()::result: ${result.length}",
+        );
+        result.sort((currentEvent, nextEvent) {
+          final currentPinnedTime =
+              currentEvent?.originServerTs ?? DateTime.now();
+          final nextPinnedTime = nextEvent?.originServerTs ?? DateTime.now();
+          return currentPinnedTime.compareTo(nextPinnedTime);
+        });
+        yield Right(ChatGetPinnedEventsSuccess(pinnedEvents: result));
+        return;
+      }
+    } on MatrixException catch (exception) {
```

Comment:
i should catch all exceptions, so that we can handle in controller, otherwise may throw at runtime

Reply:
Sure, I've taken care of it and you can see it here

<img width="301" alt="Screenshot 2023-11-07 at 11 14 22" src="https://github.com/linagora/twake-on-matrix/assets/99852347/007d62e7-dd8f-4e31-b851-72509d335ad6">